library/eos_interfaces.py

```diff
@@ -0,0 +1,282 @@
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': ['Ansible Network']}
+
+
+DOCUMENTATION = """
+---
+module: eos_interfaces
+version_added: 2.9
+short_description: Manages interface attributes of Arista EOS interfaces.
+description: ['Manages interface attributes of Arista EOS interfaces.']
+author: ['Nathaniel Case (@qalthos)']
+options:
+  config:
+    description: The provided configuration
+    suboptions:
+      description:
+        description:
+        - Interface description
+        type: str
+      duplex:
+        default: auto
+        description:
+        - Interface link status. Applicable for ethernet interface only.
+        type: str
+      enable:
+        default: true
+        description: Administrative state of the interface.
+        type: bool
+      mtu:
+        description:
+        - MTU for a specific interface. Must be an even number between 576 and 9216.
+          Applicable for Ethernet interface only.
+        type: str
+      name:
+        description:
+        - Full name of the interface, i.e. GigabitEthernet1
+        type: str
+      speed:
+        description:
+        - Interface link speed. Applicable for ethernet interface only.
+        type: str
+  state:
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    default: merged
+    description:
+    - The state the configuration should be left in
+    type: str
+
+"""
+
+EXAMPLES = """
+---
+
+# Using merged
+
+# Before state:
+# -------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Merge provided configuration with device configuration
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+        enable: True
+      - name: Ethernet2
+        description: 'Configured by Ansible'
+        enable: False
+    operation: merged
+
+# After state:
+# ------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+#    description "Configured by Ansible"
+#    shutdown
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+# Using replaced
+
+# Before state:
+# -------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Replaces device configuration of listed interfaces with provided configuration
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+        enable: True
+      - name: Ethernet2
+        description: 'Configured by Ansible'
+        enable: False
+    operation: replaced
```

Comment:
We are using `state` instead of `operation` here?

Reply:
Yeah, that fell through the model review and ended up here. Once the model is final, I need to regenerate those files and push again